app/assets/javascripts/discourse/app/components/user-menu/notification-item.js

```diff
@@ -0,0 +1,102 @@
+import GlimmerComponent from "discourse/components/glimmer";
+import { formatUsername, postUrl } from "discourse/lib/utilities";
+import { userPath } from "discourse/lib/url";
+import { setTransientHeader } from "discourse/lib/ajax";
+import { action } from "@ember/object";
+import { emojiUnescape } from "discourse/lib/text";
+import getURL from "discourse-common/lib/get-url";
+import cookie from "discourse/lib/cookie";
+import I18n from "I18n";
+
+export default class UserMenuNotificationItem extends GlimmerComponent {
+  get className() {
+    const classes = [];
+    if (this.notification.read) {
+      classes.push("read");
+    }
+    if (this.notificationName) {
+      classes.push(this.notificationName.replace(/_/g, "-"));
+    }
+    if (this.notification.is_warning) {
+      classes.push("is-warning");
+    }
+    return classes.join(" ");
+  }
+
+  get url() {
+    if (this.topicId) {
+      return postUrl(
+        this.notification.slug,
+        this.topicId,
+        this.notification.post_number
+      );
+    }
+    if (this.data.group_id) {
+      return userPath(`${this.data.username}/messages/${this.data.group_name}`);
+    }
+  }
+
+  get linkTitle() {
+    if (this.notificationName) {
+      return I18n.t(`notifications.titles.${this.notificationName}`);
+    } else {
+      return "";
+    }
+  }
+
+  get icon() {
+    return `notification.${this.notificationName}`;
+  }
+
+  get label() {
+    return this.username;
+  }
+
+  get wrapLabel() {
+    return true;
+  }
+
+  get labelWrapperClasses() {}
+
+  get username() {
+    return formatUsername(this.data.display_username);
+  }
+
+  get description() {
+    return (
+      emojiUnescape(this.notification.fancy_title) || this.data.topic_title
+    );
+  }
+
+  get descriptionElementClasses() {}
+
+  get descriptionHtmlSafe() {
+    return !!this.notification.fancy_title;
+  }
+
+  // the following props are helper props -- they're never referenced directly in the hbs template
+  get notification() {
+    return this.args.item;
+  }
+
+  get topicId() {
+    return this.notification.topic_id;
+  }
+
+  get data() {
+    return this.notification.data;
+  }
```

Comment:
I kind of prefer the more verbose version because it makes it clear that we're accessing an attribute on the `notification` object.